### PR TITLE
Add `MediaBlock`

### PR DESCRIPTION
**Pull request by SebiVPS**
Add combined media block, and remove other media blocks.

The MediaBlock itself can be used in composite blocks, the aspect ratio should be handled.
The StandaloneMediaBlock is the standalone solution with aspect ratio selection.

---

## PR Checklist

-   [x] Link to the respective task if one exists: COM-772

### CODE REVIEW

File: admin/src/common/blocks/StandaloneMediaBlock.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+export const StandaloneMediaBlock = createCompositeBlock(
+    {
+        name: "Media",
+        displayName: <FormattedMessage id="standaloneMedia.displayName" defaultMessage="Media" />,
+        blocks: {
+            media: {
+                block: MediaBlock,
+            },
+            aspectRatio: {
+                block: createCompositeSetting<StandaloneMediaBlockData["aspectRatio"]>({
```
Comment (johnnyomair) [inline admin/src/common/blocks/StandaloneMediaBlock.tsx:28]: Please use createCompositeBlockSelectField for this

File: site/src/common/blocks/DamImageBlock.tsx [hunk 2]
```diff
@@ -1,36 +1,20 @@
...
-        | { layout?: "fixed" | "intrinsic" }
-        // The sizes prop must be specified for images with layout "fill" or "responsive", as recommended in the next/image documentation
-        // https://nextjs.org/docs/api-reference/next/image#sizes
-        | {
-              layout?: "fill" | "responsive";
-              sizes: string;
-          }
-    );
+type DamImageProps = Omit<NextImageProps, "src" | "width" | "height" | "alt"> & {
+    aspectRatio?: string | "inherit";
```
Comment (johnnyomair) [inline site/src/common/blocks/DamImageBlock.tsx:7]: Aspect ratio should be required. It is required for the PixelImageBlock in the library, so it should also be required here.

File: site/src/common/blocks/MediaBlock.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    return {
+        image: (data) => <DamImageBlock data={data} sizes={sizes} aspectRatio={aspectRatio} />,
+        damVideo: (data) => <DamVideoBlock data={data} previewImageSizes={sizes} aspectRatio={aspectRatio} />,
+        youTubeVideo: (data) => <YouTubeVideoBlock data={data} previewImageSizes={sizes} aspectRatio={aspectRatio} />,
+    };
+};
+
+interface MediaBlockProps extends PropsWithData<MediaBlockData> {
+    sizes?: string;
+    aspectRatio?: string;
```
Comment (johnnyomair) [inline site/src/common/blocks/MediaBlock.tsx:15]: The aspect ratio probably needs to be required here as well

**Review by johnnyomair — APPROVED** [review]

File: site/src/documents/pages/blocks/PageContentBlock.tsx [hunk 4]
```diff
@@ -20,12 +18,7 @@ const supportedBlocks: SupportedBlocks = {
     heading: (props) => <PageContentHeadingBlock data={props} />,
     columns: (props) => <ColumnsBlock data={props} />,
     callToActionList: (props) => <PageContentCallToActionListBlock data={props} />,
-
-    // TODO: Outdated blocks! Following blocks needs refactoring to support page layout
-    image: (props) => <DamImageBlock data={props} alt="" />,
-    textImage: (props) => <TextImageBlock data={props} />,
-    damVideo: (props) => <DamVideoBlock data={props} />,
-    youTubeVideo: (props) => <YouTubeVideoBlock data={props} />,
+    media: (props) => <StandaloneMediaBlock data={props} />,
```
Comment (nsams) [inline site/src/documents/pages/blocks/PageContentBlock.tsx:21]: isn't this block missing grid styling (PageContent)? Or should it expand to full with and so needs no grid?

Comment (SebiVPS) [reply site/src/documents/pages/blocks/PageContentBlock.tsx:21]: It should expand to full width, no grid is needed 👍

**Review by mariokemetinger — APPROVED** [review]

**Review by jamesricky — APPROVED** [review]

**Review by thomasdax98 — APPROVED** [review]

**Pull request merged**